src/compas/geometry/primitives/quaternion.py

```diff
@@ -123,6 +123,40 @@ def __init__(self, w, x, y, z):
         self.y = y
         self.z = z
 
+    @property
+    def DATASCHEMA(self):
+        from schema import Schema
+        return Schema({"w": float, "x": float, "y": float, "z": float})
+
+    @property
+    def JSONSCHEMA(self):
+        from compas import versionstring
+        schema = {
```

Comment:
not to change in this PR, but shouldn't these things be loaded directly from the files instead of inlined/repeated here?

Reply:
ideally yes but somehow haven't found a builtin way for making that work

Reply:
we could add `compas.schema(...)`?

Reply:
similar to `compas.get(...)`